### PR TITLE
Rename PPOLearner to PPOPolicy and make it support continuous space

### LINKED ISSUE #92: Improve the PPOLearner to support continuous action space
Necessary changes:

1. Add a field of `dist` into the `PPOLearner` (just like @norci  did in [VPG](https://github.com/JuliaReinforcementLearning/ReinforcementLearningZoo.jl/blob/34aea9030577ed5cea6630ad02dc60e6c4cfd9a6/src/algorithms/policy_gradient/ppo.jl#L31) )

1. Following method needs to be extended to recognize environments with continuous action space. Currently the PPOLearner is assumed to return a (batch of ) logits. I'd suggest to rename the `PPOLearner` into `PPOPolicy` and return an action directly. 
https://github.com/JuliaReinforcementLearning/ReinforcementLearningZoo.jl/blob/34aea9030577ed5cea6630ad02dc60e6c4cfd9a6/src/algorithms/policy_gradient/ppo.jl#L77

1. [GausianNetwork](https://github.com/JuliaReinforcementLearning/ReinforcementLearningZoo.jl/pull/87/files#diff-dc9b8ba63840800093295c526ec3aa91R9-R13) is also needed.

1. Calculating entropy loss in `update!` is hard coded. Better to split it into a function to support continuous distribution (or reuse the one in StatsBase or Distributions. But use them with caution! I had some problems using them with Zygote before)

### PR DESCRIPTION
**Pull request by findmyway**
Fix #92 

- [x] Need a new release from RLCore.
- [x] Need to improve RamdomPolicy in RLBase to support MultiThreadEnv
- [x] Add a test case.

### CODE REVIEW

**Comment by findmyway** [comment]
I've test PPO related experiments. It seems everything works fine (both on Pendulum/Atari) except that the speed of ``E`rlpyt_PPO_Atari(Pong)` `` decreased from `711fps` to about `590fps`.

Not sure why it happens. @norci @stasti @rbange  Since you may be interested in this PR, suggestions are welcomed.

**Comment by rbange** [comment]
The performance decrease is kinda drastic... unfortunately im only working on continuous action space problems, so I have no deep insights into the Atari domain... My first guess would be some performance flaw in `function (agent::Agent{<:Union{PPOPolicy, RandomStartPolicy{<:PPOPolicy}}})(::Training{PreActStage}, env::MultiThreadEnv)` or `get_prob` as these seem to be the only performance relevant changes. But from a code review standpoint... idk.

I would like to encourage you in moving the Gaussian Network outside of the VPG algorithm, as this kind of network can be used in the Soft Actor Critic as well as in the PPO now. There are a few issues regarding the definition and redundancy of specific neural network architectures and patterns. As this implementation works in here I guess ill better open an issue to document my thoughts regarding this.

File: src/algorithms/policy_gradient/ppo.jl [hunk 1]
```diff
@@ -142,60 +161,53 @@ function RLBase.update!(learner::PPOLearner, t::PPOTrajectory)
...
+function (agent::Agent{<:Union{PPOPolicy, RandomStartPolicy{<:PPOPolicy}}})(::Training{PreActStage}, env::MultiThreadEnv)
     state = get_state(env)
+    dist = get_prob(agent.policy, env)
+    if dist isa Matrix{<:Number}
+        dist = [Categorical(x;check_args=false) for x in eachcol(dist)]
+    elseif dist isa Vector{<:Vector{<:Number}}
+        dist = [Categorical(x;check_args=false) for x in dist]
+    end
+    action = agent.policy(env)
+    action_log_prob = [logpdf(d, a) for (d, a) in zip(dist, action)]
```
Comment (stasti) [inline src/algorithms/policy_gradient/ppo.jl:220]:
```suggestion
    action_log_prob = logpdf.(dist, action)
```

Comment (stasti) [reply src/algorithms/policy_gradient/ppo.jl:220]: a bit faster

File: src/algorithms/policy_gradient/ppo.jl [hunk 2]
```diff
@@ -74,21 +81,33 @@ function PPOLearner(;
...
+        state,
+    )) |> send_to_host |> StructArray{Normal}
+end
+
+function RLBase.get_prob(p::PPOPolicy{<:ActorCritic, Categorical}, state::AbstractArray)
+    logits = p.approximator.actor(send_to_device(
+        device(p.approximator),
+        state,
+    )) |> softmax |> send_to_host
+    [Categorical(x;check_args=false) for x in eachcol(logits)]
```
Comment (stasti) [inline src/algorithms/policy_gradient/ppo.jl:96]: to move all '[...]'  to '@.'  ?

**Comment by norci** [comment]
I tried to run the PPO Atari experiment, but I'm not able to build `ArcadeLearningEnvironment`, due to network issue.
```
Error: Download failed: curl: (7) Failed to connect to raw.githubusercontent.com port 443: Connection refused
```

For performance issue, could you have a look at
* https://docs.julialang.org/en/v1/manual/profile/
* https://github.com/davidanthoff/ProfileVega.jl#differential-flame-graphs ( ... is a useful tool for comparison code before and after changes.)

**Comment by findmyway** [comment]
Thanks to @rbange 's comments. I figured out the problem lied in that the `get_prob` was called twice. One for calculating distribution, one for calculating the actions. The later can be simply replaced with sampling from the former resulted distribution.

<hr>

> I tried to run the PPO Atari experiment, but I'm not able to build ArcadeLearningEnvironment, due to network issue.

@norci  You may need （翻）VPN（墙） 😉
